File: projects-refactored/ecommerce-api-legacy/src/middlewares/errorHandler.js
function errorHandler(err, req, res, next) {
  if (err.name === 'NotFoundError') return res.status(404).json({ error: err.message });
  if (err.name === 'ValidationError') return res.status(422).json({ error: err.message });
  if (err.name === 'AuthError') return res.status(401).json({ error: err.message });
  if (err.name === 'PaymentError') return res.status(400).json({ error: err.message });
  console.error('[ERROR]', err.message);
  res.status(500).json({ error: 'Internal server error' });
}

module.exports = errorHandler;

File: projects-refactored/task-manager-api/middlewares/error_handler.py
import logging

logger = logging.getLogger(__name__)


class NotFoundError(Exception):
    pass


class ValidationError(Exception):
    pass


class AuthError(Exception):
    pass


def register_error_handlers(app):
    from flask import jsonify

    @app.errorhandler(NotFoundError)
    def handle_not_found(e):
        return jsonify({'error': str(e)}), 404

    @app.errorhandler(ValidationError)
    def handle_validation(e):
        return jsonify({'error': str(e)}), 422

    @app.errorhandler(AuthError)
    def handle_auth(e):
        return jsonify({'error': str(e)}), 401

    @app.errorhandler(Exception)
    def handle_generic(e):
        logger.exception('Unhandled error')
        return jsonify({'error': 'Internal server error'}), 500
